### DISCUSSION TITLE
Using file filters inside configuration to import list of domains for rules

### DISCUSSION THREAD

**Discussion by sarpuser** [authelia/authelia#7329] [Q&A] [answered]
Would it be possible to use a file filter to import the list of domains for a given rule in the configuration file? It is annoying to sroll through the config file every time I add a new domain and want to add it to one either the two factor or the one factor rules. How would I set this up?

**Comment by glime-ai** [comment]
Please note this response was created using generative AI leveraging previously available information from the repository. This information may be helpful however it is our strong recommendation that you carefully consider the information before blindly applying it.

---

Yes, it is possible to use a file filter to manage the list of domains dynamically in Authelia's configuration file. Authelia supports the use of file filters which can be applied to configuration files to dynamically adjust settings based on external inputs or environment variables.

To set this up, you can utilize the "Expand Environment Variable Filter" or the "Go Template Filter" as mentioned in the Authelia documentation. Here’s a general approach on how you might configure this:

1. **Prepare Your Domain List File**: Create a separate file that contains your list of domains. You can structure this file in a simple format that lists one domain per line or any other format that suits your needs.

2. **Use Environment Variables**: If you choose the "Expand Environment Variable Filter", you can set an environment variable that points to your domain list file or includes the domains directly.

3. **Modify Authelia Configuration**: In your `configuration.yml` or the specific configuration file you are using, apply a file filter that reads from your domain list file or the environment variable. You might use a Go Template to iterate over the domains and dynamically insert them into the configuration.

4. **Reload Configuration**: Ensure that Authelia reloads the configuration or restart the service to apply the new settings.

Here is a basic example of how you might define this in a configuration file using a Go Template Filter:

```yaml
access_control:
  rules:
    - domain: "{{ range $index, $domain := .Env.DOMAIN_LIST }}{{ if $index }},{{ end }}{{ $domain }}{{ end }}"
      policy: two_factor
```

In this example, `DOMAIN_LIST` would be an environment variable containing a comma-separated list of domains. You would need to ensure that this environment variable is properly populated and exported in the environment where Authelia is running.

Remember, the exact implementation can vary based on your specific setup and requirements. It’s important to test any changes in a development environment before applying them in production.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=6fsHkcyufjQLaB&utm_medium=github).

**Comment by james-d-elliott** [comment, marked as answer]
The env as suggested is probably the best way. We have no way at this stage to import values files like helm does, but we may add it in the future. Sorry for not getting back sooner.
